sanic_ext/extensions/injection/registry.py

```diff
@@ -22,8 +22,8 @@ def get(self, key, default=None):
     def register(
         self, _type: Type, constructor: Optional[Callable[..., Any]]
     ) -> None:
-        if constructor:
-            constructor = Constructor(constructor)
+        constructor = constructor or _type
```

Comment:
This is done in the `do_cast` method already https://github.com/sanic-org/sanic-ext/pull/128/files#diff-bd7bd8266dcd701b4f963cb7a8f2b8d34aa9b8777fe85c5301f9c369d4d74fe4R165